keras_nlp/tokenizers/byte_pair_tokenizer_test.py

```diff
@@ -66,6 +65,25 @@ def test_tokenize_string_output(self):
         )
         self.assertAllEqual(call_output, expected)
 
+    def test_tokenize_with_special_tokens(self):
```

Comment:
have you tested detokenization? That is probably worth checking out

Reply:
yea, it's actually not affected by this PR. 

Here the special token has to be present inside the vocab, such as `<|endoftext|>`.